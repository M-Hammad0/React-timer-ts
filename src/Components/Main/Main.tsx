import React, { useState, useEffect } from 'react'
import './Main.css'
import CalculateTimer from './../../CalculateTimer';
import Controls from '../Controls/Controls';

const Main = () => {
      const [timeInSeconds, setTimeInSeconds] = useState<number>(0);
      const [timerArray, setTimerArray] = useState<Array<number|string>>([]);

      useEffect(() => {
           let timeArray: Array<number|string> = CalculateTimer(timeInSeconds);
           setTimerArray(timeArray);
      },[timeInSeconds])

      return (
            <div>
            <div className="time-container">
                  <p className="timer-text">{timerArray[0]}</p>
                  <span>:</span>
                  <p className="timer-text">{timerArray[1]}</p>
                  <span>:</span>
                  <p className="timer-text">{timerArray[2]}</p>
            </div>
            <Controls time={setTimeInSeconds}/>
            </div>
      )
}

export default Main
