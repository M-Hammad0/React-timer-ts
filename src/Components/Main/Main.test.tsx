import React from 'react'
import '@testing-library/jest-dom'
import { render, cleanup} from '@testing-library/react';
import Main from './Main';

afterEach(cleanup)

it('render Main component', () => {
      const {asFragment} = render(<Main />);
      expect(asFragment).toMatchSnapshot();
})

it('render correct time',  () => {
      const {getAllByText} =   render(<Main />);
      expect(getAllByText('00')).toHaveLength(3)
})
