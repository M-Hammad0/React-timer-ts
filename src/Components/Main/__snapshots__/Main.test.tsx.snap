// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Main component 1`] = `[Function]`;
